src/parse/body.rs

```diff
@@ -0,0 +1,465 @@
+use std::io::BufRead;
+
+use crate::common::header::{HeaderMap, HeaderMapOps};
+use crate::common::header;
+use crate::parse::body::BodyParser::{Chunked, Empty, UntilEof, WithSize};
+use crate::parse::body::chunked::ChunksParser;
+use crate::parse::deframe::bytes::BytesDeframer;
+use crate::parse::deframe::bytes_until_eof::BytesUntilEofDeframer;
+use crate::parse::deframe::deframe::Deframe;
+use crate::parse::error::ParsingError;
+use crate::parse::error_take::ReadExt;
+use crate::parse::parse::{Parse, ParseResult};
+use crate::parse::parse::ParseStatus::Done;
+
+/// The maximum size of a body.
+const MAX_BODY_SIZE: usize = 3 * 1024 * 1024; // 3 megabytes
```

Comment:
Why 3mb?

Reply:
Good question haha. I think I went to some random websites and the response sizes were never greater than 3mb. It might make sense to make this configurable through the config though.

Reply:
Yeah that's what I was thinking